File: 1005-maximize-sum-of-array-after-k-negations/1005-maximize-sum-of-array-after-k-negations.cpp
class Solution {
public:
    int largestSumAfterKNegations(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> minHeap;
        
        for (int num : nums) minHeap.push(num);
        
        while (k > 0){
            int n = minHeap.top();
            minHeap.pop();
            minHeap.push(-n);
            --k;
        }
        
        int ans = 0;
        while (minHeap.size() > 0) {
            ans += minHeap.top();
            minHeap.pop();
        }
        
        return ans;
    }
};

File: 1005-maximize-sum-of-array-after-k-negations/1005-maximize-sum-of-array-after-k-negations.py
class Solution:
    def largestSumAfterKNegations(self, nums: List[int], k: int) -> int:
        heap = []
        
        for num in nums:
            heappush(heap, num)
        
        while k > 0:
            elem = heappop(heap)
            heappush(heap, -elem)
            k -= 1
        
        return sum(heap)